sympy/core/relational.py

```diff
@@ -311,8 +311,14 @@ def _eval_simplify(self, **kwargs):
             v = None
             if dif.is_comparable:
                 v = dif.n(2)
-            elif dif.equals(0):  # XXX this is expensive
-                v = S.Zero
+            else:
+                from sympy.sets import Set
+                if not isinstance(dif, Set):
+                    if dif.equals(0):  # XXX this is expensive
```

Comment:
I don't think that sets should be a special case for sets here. There needs to be some kind of check here but it shouldn't be a `if not` negative check. The sets module should not be imported here in core.